components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -601,6 +601,20 @@ mojom::APIError ConversationDriver::GetCurrentAPIError() {
   return current_error_;
 }
 
+mojom::ConversationTurnPtr ConversationDriver::ClearErrorAndGetFailedMessage() {
+  DCHECK(!chat_history_.empty());
+
+  SetAPIError(mojom::APIError::None);
+  mojom::ConversationTurnPtr turn = chat_history_.back().Clone();
+  chat_history_.pop_back();
```

Comment:
instead of cloning, can't we just move the entry, since we're getting rid of it anyway?

Reply:
yes, you can, but before sending it to the IPC, you'd have to turn it into a `mojo::StructPtr<T>` so, there's no point in returning a movable type here because we will end up cloning anyways.

Reply:
note to self: put a todo

Reply:
ok, let's put a TODO in the header file for the vector class member